docs/migration-guides/prometheus-migration.md

```diff
@@ -0,0 +1,328 @@
+
+```bash
+~ $ kubectl get svc
+NAME               TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
+prometheus         ClusterIP   10.30.244.231   <none>        9090/TCP   1y
+prometheus-rsync   ClusterIP   10.30.247.115   <none>        873/TCP    9d
+```
+
+Now, you should be able to access to that service from other pod. If you are interested in doing a quick test from other pod you can do:
+
+```bash
+/ # nc -zv prometheus-rsync.NAMESPACE.svc.cluster.local 873
+prometheus-rsync.NAMESPACE.svc.cluster.local (10.130.32.5:873) open
+```
+
+## Modifying the `kubeprod-manifest.jsonnet` to add the sidecar rsync container
+
+The `kubeprod-manifest.jsonnet` file will contain all the custom configuration you want to apply to the jsonnet manifests.  In this case, in order to add the rsync sidecar container the following configuration will have to be added:
+
+```jsonnet
+// Bitnami specific common configuration
+local common_prom_config = import "../common/prometheus/conf.d/prometheus.jsonnet";
+
+// Common libs
+local kube = import "kube.libsonnet";
+
+(import "../vendor/kubeprod/manifests/platforms/gke.jsonnet") {
+
+  config:: import "kubeprod-autogen.json",
+
+  // Place your overrides here
+  prometheus+: { 
+
+    // Add rsync sidecar container to the prometheus sts.
+    // and adapt resources to our needs
+    prometheus+: {  // Prometheus server itself
+      deploy+: {  // Prometheus StatefulSet deployment
+        spec+: {
+          template+: {
+            spec+: {
+              containers_+: {
+                rsync: kube.Container("sidecar-rsync") {
+                  image: "alpine:3.8",
+                  args+: ["sh", "-c", "apk update && apk add rsync curl && tail -f /dev/null"],
+                  volumeMounts_+: {
+                    data: { mountPath: "/data" },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+}
+```
+
+After you deploy the manifests in your Kubernetes cluster, you will see the 3 containers that the Prometheus pod has:
+
+```bash
+~ $ kubectl get pods -o=jsonpath='{range .items[*]}{"\n"}{.metadata.name}{":\t"}{range .spec.containers[*]}{.image}{", "}{end}{end}' | grep prometheus |\
+sort
+
+prometheus-0:	bitnami/prometheus:2.4.3-r31, jimmidyson/configmap-reload:v0.2.2, alpine:3.8,
+```
+
+Now, you can do a quick test of the rsync service by getting a shell in the container executing:
+
+`kubectl exec -it prometheus-0 -n kubeprod -c sidecar-rsync sh`
+
+and:
+
+`rsync -avrP prometheus-rsync.NAMESPACE.svc.cluster.local::data/test-file .`
+
+> Make sure previously you have deployed a test file in your Prometheus data directory so you can rsync it to the new container
+
+## Syncing with the compact times of the Prometheus database
+Now that all bits are in place you are ready to perform the Prometheus TSDB migration. To do so, you will first have to sync with the TSDB compact times of Prometheus to ensure the minimal data loss.
```

Comment:
After reading this step, I realized that this step focuses more on explaining the concept of the overlap and the commands are in the next step. So i would modify the setup sentences as "To do so, you will first have to understand how Prometheus' compaction process works and how to sync compaction times to minimize data loss." or similar

Reply:
done, thanks.